sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/Configs.java

```diff
@@ -494,4 +498,10 @@ public static Duration getTcpConnectionAcquisitionTimeout(int defaultValueInMs)
             )
         );
     }
+
+    public static boolean shouldDiagnosticsProviderSystemExitOnError() {
+        return getJVMConfigAsBoolean(
```

Comment:
please use the SystemProperty with fallback to env variable and fallback to default pattern.

Reply:
Updated